Tent.js

```diff
@@ -0,0 +1,98 @@
+const assignIn = require('lodash.assignin');
+const cloneDeep = require('lodash.clonedeep');
+const fs = require('fs');
+const merge = require('lodash.merge');
+const path = require('path');
+const pick = require('lodash.pick');
+const semver = require('semver');
+const shell = require('shelljs');
+const tmp = require('tmp');
+
+const manifestJSON = require('./manifest.json');
+const packageJSON = require('./package.json');
+
+shell.set('-e');
+
```

Comment:
I'm going to document the external API in a separate PR once #38 is landed, let's get this in first!